src/index.js

```diff
@@ -1 +1,56 @@
 'use strict';
+
+const http = require('http');
+const zlib = require('zlib');
+
+const server = http.createServer((req, res) => {
+  if (req.method === 'POST' && req.url === '/compress') {
+    let fileToCompress = Buffer.alloc(0);
+
+    req.on('data', chunk => {
+      fileToCompress = Buffer.concat([fileToCompress, chunk]);
+    });
+
+    req.on('end', () => {
+      let compression;
+      const fields = {
```

Comment:
Consider using the request object to get the compression type selected by the user